### PR TITLE
203 | Add unit testing documentation and examples

### LINKED ISSUE #203: Improve and document unit tests best practices
## Overview

We want to have best practices for unit testing well documented. The goal of this issue is to create one good example of this and the documentation with the guidelines.

## Acceptance criteria

- Improve at least one component testing
- Define unit testing best practices that will be adopted
- Document best practices
- Exemplify best practices with an existing component
- Include bad examples as well

## Tasks

- [ ] Define best practices for unit tests
- [ ] Review the chosen best practices with the team in a meeting
- [ ] Create the document with good and bad examples of the good practices
- [ ] Create the chosen component tests

### PR DESCRIPTION
**Pull request by EJSG**
# Fixes #203 

## Description

This PR adds in some example unit tests, documentation, and also fixes staking input v-model issues found during test creation

## Test scenarios

**Regression**
- go to http://localhost:8080/staking
- type various values in the staking inputs
    - there should be no perceptible change from prod

## Checklist:

-   [x] I have performed a self-review of my own code
-   [x] I have commented my code, particularly in hard-to-understand areas
-   [x] I have cleaned up the code in the areas my change touches
-   [x] My changes generate no new warnings
-   [x] Any dependent changes have been merged and published in downstream modules
-   [x] I have checked my code and corrected any misspellings
-   [x] I have removed any unnecessary console messages

### CODE REVIEW

File: src/pages/staking/StakingFormInput.vue [hunk 1]
```diff
@@ -75,10 +75,10 @@ const dotRegex = /\./g;
 export default {
     name: 'StakingFormInput',
     props: {
-        value: {
+        modelValue: {
```
Comment (EJSG) [inline src/pages/staking/StakingFormInput.vue:78]: vmodel implementation was done in a vue2 style; has been updated for v3

File: test/__mocks__/svg.mocks.js [hunk 2]
```diff
@@ -0,0 +1,9 @@
+module.exports = {
+    process() {
+        return { code: 'module.exports = {};' };
+    },
+    getCacheKey() {
+        // The output is always the same.
+        return 'svgTransform';
+    },
+};
```
Comment (EJSG) [inline test/__mocks__/svg.mocks.js:9]: allow jest to mock SVG files

**Review by donnyquixotic — APPROVED** [review]
@EJSG approved as-is but here's documentation on adding jest via Quasar extension installer if you want to try it out first before merging: https://quasar.dev/quasar-cli-webpack/testing-and-auditing#documentation It may simplify stubbing quasar components but I have not tested snapshot testing with it yet.

**Review by karynemayer — APPROVED** [review]

File: docs/Testing.md [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+    ```
+
+
+### Snapshots <a name="#snapshots"></a>
+Snapshots are one of the most important tools in the testing toolset. They allow one to verify the state of the rendered
+markup (as well as other things) deterministically. Rather than manually checking that a stub component or an element
+has specific props, you can call `expect(wrapper.element).toMatchSnapshot()`
+
+Two things to keep in mind:
+1. it is extremely important that you check generated snapshots for validity _manually_ every time they are created or
```
[the remark this answers is not in the document]
Comment (EJSG) [reply docs/Testing.md:91]: not that I can see, in this case I wanted "manually" to be italic

**Review by karynemayer — APPROVED** [review]

**Review by donnyquixotic — APPROVED** [review]

**Pull request merged**